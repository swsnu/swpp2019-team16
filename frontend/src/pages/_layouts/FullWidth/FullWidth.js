import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../../../lib/styles/theme';
import ThemeProvider from '@material-ui/styles/ThemeProvider';
import { CssBaseline } from '@material-ui/core';

const Root = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
  background: ${theme.palette.background.default};

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

FullWidth.propTypes = {
  children: PropTypes.node.isRequired,
};

function FullWidth({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {/* <CssBaseline /> enables lib/material-ui/theme to work */}
      <CssBaseline />
      <Root>{children}</Root>
    </ThemeProvider>
  );
}

export default FullWidth;
