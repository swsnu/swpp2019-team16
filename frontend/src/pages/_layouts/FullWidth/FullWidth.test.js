import React from 'react';
import { render } from '@testing-library/react';
import FullWidth from './FullWidth';

describe('<FullWidth />', () => {
  it('SHOULD match with snapshoot', async () => {
    const { container } = render(
      <FullWidth>
        <div>Full width page</div>
      </FullWidth>,
    );
    expect(container).toMatchSnapshot();
  });
});
