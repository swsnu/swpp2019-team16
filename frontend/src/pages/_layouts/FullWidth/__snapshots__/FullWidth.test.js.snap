// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullWidth /> SHOULD match with snapshoot 1`] = `
<div>
  <div
    class="sc-bdVaJa fJyqCR"
  >
    <div>
      Full width page
    </div>
  </div>
</div>
`;
